src/shared/quota-util.c

```diff
@@ -33,3 +34,61 @@ int quotactl_fd_with_fallback(int fd, int cmd, int id, void *addr) {
 
         return RET_NERRNO(quotactl(cmd, devnode, id, addr));
 }
+
+int quota_query_proj_id(int fd, uint32_t proj_id, struct dqblk *ret_req) {
+        int r;
+
+        assert(fd >= 0);
+        assert(ret_req);
+
+        r = RET_NERRNO(quotactl_fd(fd, QCMD_FIXED(Q_GETQUOTA, PRJQUOTA), proj_id, ret_req));
+        if (r == -ESRCH || ERRNO_IS_NEG_NOT_SUPPORTED(r) || ERRNO_IS_NEG_PRIVILEGE(r))
+                return false;
```

Comment:
we have this rule in systemd that return params should either be fully initialized on success or untouched on failure. Failure is indicated by return values < 0, success by those >= 0. This means here you return "false", which is a kind of success here. Hence you should initialize `ret_req` fully.